app/main.py

```diff
@@ -1,2 +1,74 @@
+from typing import Any, Hashable, Optional
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.hash_table: list = [None] * 8
+        self.capacity = 8
+
+    def get_index_from_hash(self, key: Hashable) -> int:
+        return hash(key) % self.capacity
+
+    def __setitem__(
+            self,
+            key: Hashable,
+            value: Any
+    ) -> None:
+        if self.length >= self.capacity * (2 / 3):
+            self.resize()
+        index = hash(key) % self.capacity
+        while True:
+            if not self.hash_table[index]:
+                self.length += 1
+            if not self.hash_table[index] or self.hash_table[index][0] == key:
+                self.hash_table[index] = (key, hash(key), value)
+                break
+            index = (index + 1) % self.capacity
+
+    def __getitem__(self, key: Hashable) -> Any:
+        return self.get(key)
+
+    def get_index(self, key: Hashable) -> int | KeyError:
+        start_index = hash(key) % self.capacity
+        index = start_index
+        while True:
+            if self.hash_table[index] and self.hash_table[index][0] == key:
+                return index
+            index = (index + 1) % self.capacity
+            if index == start_index:
+                raise KeyError("Key was not found")
+
+    def get(self, key: Hashable, default: Optional[Any] = None) -> Any:
+        try:
+            return self.hash_table[self.get_index(key)][2]
+        except KeyError:
+            if default is not None:
+                return default
+            raise KeyError(key)
```

Comment:
`get` should not raise a `KeyError` in any case, it should return `None` when the `default` is not provided since `None` is a default option.